starfish/expression_matrix/expression_matrix.py

```diff
@@ -0,0 +1,146 @@
+import anndata
+import loompy
+import xarray as xr
+
+
+class ExpressionMatrix(xr.DataArray):
+    """Container for expression data extracted from an IntensityTable
+
+    An ExpressionMatrix is a 2-dimensional ``cells`` x ``genes`` tensor whose values are the
+    number of spots observed for each gene observed by the experiment. In addition to the basic
+    xarray methods, IntensityTable implements:
+
+    Methods
+    -------
+    save(filename)
+        save the ExpressionMatrix to netCDF
+
+    save_loom(filename)
+        save the ExpressionMatrix to loom for use in R or python
+
+    save_anndata(filename)
+        save the ExpressionMatrix to AnnData for use in ``Scanpy``
+
+    load(filename)
+        load an ExpressionMatrix from netCDF
+
+    load_loom(filename)
+        load an ExpressionMatrix from loom
+
+    load_anndata(filename)
+        load an ExpressionMatrix from AnnData
+
+
+    Examples
+    --------
+    # TODO ambrosejcarr write examples
+
+    """
+
+    def to_anndata(self) -> anndata.AnnData:
+        """convert ExpressionMatrix to a scanpy-compatible AnnData object"""
+        row_attrs = {k: self['cells'][k].values for k in self['cells'].coords}
+        col_attrs = {k: self['genes'][k].values for k in self['genes'].coords}
+        return anndata.AnnData(self.data, row_attrs, col_attrs)
+
+    def save(self, filename: str) -> None:
+        """Save an ExpressionMatrix as a Netcdf File
+
+        Parameters
+        ----------
+        filename : str
+            Name of Netcdf file
+
+        """
+        self.to_netcdf(filename)
+
+    def save_loom(self, filename: str) -> None:
+        """Save an ExpressionMatrix as a loom file
+
+        Parameters
+        ----------
+        filename : str
+            Name of loom file
+
+        """
+        row_attrs = {k: self['cells'][k].values for k in self['cells'].coords}
+        col_attrs = {k: self['genes'][k].values for k in self['genes'].coords}
+
+        loompy.create(filename, self.data, row_attrs, col_attrs)
+
+    def save_anndata(self, filename: str) -> None:
+        """Save an ExpressionMatrix as an AnnData file
+
+        Parameters
+        ----------
+        filename : str
+            Name of AnnData file
+
+        """
+        self.to_anndata().write(filename)
+
+    @classmethod
+    def load(cls, filename: str) -> "ExpressionMatrix":
+        """load an ExpressionMatrix from Netcdf
+
+        Parameters
+        ----------
+        filename : str
+            File to load
+
+        Returns
+        -------
+        ExpressionMatrix
+
+        """
+        loaded = xr.open_dataarray(filename)
+        expression_matrix = cls(
+            loaded.data,
+            loaded.coords,
+            loaded.dims
+        )
```

Comment:
Could this load method detect what kind of file (Netcdf, AnnData, loom) is being passed in and respond accordingly? So you'd only need one load method?

Reply:
I'm not sure I can trust AnnData and Loom have used magic numbers in their formats. It looks like we'd need to open the file with pytables to inspect the index to try to guess which is which. That could work. More likely we'd need to base it off of file extensions, which I don't really like to do. 

I foresee such a function ending with "Warning: could not autodetect file format, defaulting to `netCDF`. To override, please use a specific `load_<format>` method". 

At least in the short term, I don't know if this is worth the time. Honestly I expect most people to take their matrix, leave, and never return to starfish. :-)